src/Sql/Sql/Data Sync/Cmdlet/NewAzureSqlSyncGroup.cs

```
@@ -149,11 +162,48 @@ protected override IEnumerable<AzureSqlSyncGroupModel> ApplyUserInputToModel(IEn
                 ServerName = this.ServerName,
                 DatabaseName = this.DatabaseName,
                 SyncGroupName = this.Name,
-                ConflictResolutionPolicy = this.ConflictResolutionPolicy != null ? this.ConflictResolutionPolicy.ToString() : null,
-                HubDatabaseUserName = this.DatabaseCredential != null ? this.DatabaseCredential.UserName : null,
-                HubDatabasePassword = this.DatabaseCredential != null ? this.DatabaseCredential.Password : null
+                ConflictResolutionPolicy = this.ConflictResolutionPolicy != null ? this.ConflictResolutionPolicy.ToString() : null
             };
 
+            if (!MyInvocation.BoundParameters.ContainsKey(nameof(HubDatabaseAuthenticationType)) ||
+                this.HubDatabaseAuthenticationType.Equals("password", System.StringComparison.OrdinalIgnoreCase))
+            {
+                if (!MyInvocation.BoundParameters.ContainsKey(nameof(DatabaseCredential)) ||
+                    this.DatabaseCredential == null ||
+                    string.IsNullOrEmpty(this.DatabaseCredential.UserName))
+                {
+                    throw new PSArgumentException(
+                        Microsoft.Azure.Commands.Sql.Properties.Resources.DatabaseCredentialRequired,
+                        "DatabaseCredentials");
+                }
+
+                newModel.HubDatabaseUserName = this.DatabaseCredential.UserName;
```

At this point, if parameter `IdentityId` is supplied in the input, the code should reject the input rather than ignoring it and going ahead with username and password. This is dependent parameter validation with the goal of rejecting requests that are badly formed.